tests/bp/test_infer.py

```diff
@@ -0,0 +1,119 @@
+from itertools import product
+
+import numpy as np
+
+from pgmax.bp import infer
+
+print("We assume that factor_id is increasing and take values in 0,..., n_factors")
```

Comment:
We should be able to incorporate this assumption when we compile various arrays. Just need to be a bit careful about how to structure the interface but the assumption should be fine